wrap.go

```diff
@@ -67,9 +67,8 @@ func DecorateMany(message string, errs ...error) error {
 
 	if areAllOfTheSameType(errs...) {
 		return WrapMany(transparentWrapper, message, errs...)
-	} else {
```

Comment:
There is a thing here that annoys me a little. Here's a guideline we are trying to comply with here: 
https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow
And it goes well when there is a 'negative' branch (greater indentation) and a 'positive' branch (minimal indentation). 
Here, on the other hand, there a 2 positive branches, so it seems that changing the code in this manner actually makes it send a wrong message to a reader.
Thoughts?

As for other similar fixes in this PR, I suggest we apply the same rule. If we follow this rule, at lest it should show what is considered a 'positive' codepath.

Reply:
You bring up a good point. In the example you highlighted I would argue that `!areAllOfTheSameType` would be the negative, so the logic should be reversed. As for the others I think the ones in `builder.go` could also be flipped. Final choice is yours. If you prefer to leave these I can remove them from the PR and just leave the doc fix. Thoughts?

Reply:
The main point is whether or not *linter* is being a little too zealous here. I originally felt like it is, and that's why those reports remained unsolved until now. But if you feel otherwise, I'm ready to reconsider.
Could you please modify this PR so that this rule of positive path is best respected, and then I'll make comments if I disagree with some of those choices?